### DISCUSSION TITLE
creating new listing : Error adding listing: BadRequest - {"error":"IDs must be >= 1, got 0"}

### DISCUSSION THREAD

**Discussion by digital9999999** [etsy/open-api#1251] [Q&A]
(https://developers.etsy.com/documentation/reference/#operation/createDraftListing)

  var newListing = new Listing
  {
      shop_id = 52654032,
      Title = "Handmade Bracelet",
      Description = "A beautiful handmade bracelet.",
      Price = 25,
      Quantity = 10,
      TaxonomyId = 0,
      whomade = "i_did",
      listingtype = "digital",
      whenmade = "2020_2024",
      state = "draft"

  };
  returns error: Error adding listing: BadRequest - {"error":"IDs must be >= 1, got 0"}
  are we missing anything?

**Comment by mjec** [comment]
`taxonomy_id` must be an integer >= 1, as noted in the docs you linked, but it looks like you're using `0`. See [SellerTaxonomy](https://developers.etsy.com/documentation/reference#tag/SellerTaxonomy) and [BuyerTaxonomy](https://developers.etsy.com/documentation/reference#tag/BuyerTaxonomy) for more information.

**Reply by digital9999999** [reply]
same error - tried a few different TaxonomyId's

anyone have a working c# request they can share? 

 var newListing = new Listing
 {
     shop_id = shopId,
     Title = "Handmade Bracelet",
     Description = "A beautiful handmade bracelet.",
     Price = 25,
     Quantity = 10,
     TaxonomyId = 1,
     whomade = "i_did",
     listingtype = "digital",
     whenmade = "2020_2024",
     state = "draft",
     is_supply= "false",
 };
      
 try
 {
     var response = await etsyApiClient.AddListingAsync(newListing);
     Console.WriteLine($"Listing added successfully: {response}");
 }
 catch (Exception ex)
 {
     Console.WriteLine($"Error: {ex.Message}");
 }
<img width="480" alt="error" src="https://github.com/etsy/open-api/assets/172336310/6d387201-38ea-461d-bda3-20efd2b1b709">

**Comment by slider1234** [comment]
Add debugging lines before making the API call to ensure that all IDs are set correctly. Here is the line you need to see if your ouput is correct before it goes to the API. 

`Console.WriteLine($"shop_id: {newListing.shop_id}, TaxonomyId: {newListing.TaxonomyId}");`
